contrib/babelfishpg_tsql/src/procedures.c

```diff
@@ -1588,6 +1590,220 @@ create_xp_instance_regread_in_master_dbo_internal(PG_FUNCTION_ARGS)
+					}
+				}
+
+				if(!flag)
+				{
+					ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+					 		errmsg("'%s' extension creation is not supported", crstmt->extname)));
+				}
+
+				if (!superuser_arg(GetSessionUserId()) || !role_is_sa(GetSessionUserId()))
+				{
+					ereport(ERROR,
+						(errcode(ERRCODE_INSUFFICIENT_PRIVILEGE),
+	 					errmsg("permission denied to create extension")));
+				}
+
+				SetCurrentRoleId(GetSessionUserId(), false);
+				SetConfigOption("search_path",
+								new_path,
+								PGC_SUSET,
+								PGC_S_DATABASE_USER);
+
+				foreach(lc, crstmt->options)
+				{            
+					DefElem    *defel = (DefElem *) lfirst(lc);                                                                                                                                                                               
+					if (strcmp(defel->defname, "schema") == 0) 
+					{
+						d_schema = defel;
+						schemaName = defGetString(d_schema);
+					}
+					if (strcmp(defel->defname, "cascade") == 0)
+					{
+						ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+							errmsg("'cascade' is not yet supported in Babelfish")));
+					}
+				}
+
+				if(schemaName != NULL && !(is_shared_schema(schemaName)))
+				{
+					ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+							errmsg("'%s' is not a valid schema", schemaName)));
+				}
+
+				/* do this step */
+				ProcessUtility(wrapper,
+							postgresStmt,
+							false,
+							PROCESS_UTILITY_SUBCOMMAND,
+							NULL,
+							NULL,
+							None_Receiver,
+							NULL);
+
+				/* make sure later steps can see the object created here */
+					CommandCounterIncrement();
+				break;
+			}
+			case T_DropStmt:
+			{
+				DropStmt *drstmt = (DropStmt *) parsetree;
+
+				SetCurrentRoleId(GetSessionUserId(), false);
+
+				if(drstmt->behavior == DROP_CASCADE)
+				{
+					ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+								errmsg("'cascade' is not yet supported in Babelfish")));
+				}
+
+				if (drstmt->removeType == OBJECT_EXTENSION)
+				{
+					/* do this step */
+					ProcessUtility(wrapper,
+								postgresStmt,
+								false,
+								PROCESS_UTILITY_SUBCOMMAND,
+								NULL,
+								NULL,
+								None_Receiver,
+								NULL);
+
+					/* make sure later steps can see the object created here */
+					CommandCounterIncrement();
+				}
+				break;
+			}
+			case T_AlterExtensionStmt:
+			{
+				SetCurrentRoleId(GetSessionUserId(), false);
+				/* do this step */
+				ProcessUtility(wrapper,
+							postgresStmt,
+							false,
+							PROCESS_UTILITY_SUBCOMMAND,
+							NULL,
+							NULL,
+							None_Receiver,
+							NULL);
+
+				/* make sure later steps can see the object created here */
+				CommandCounterIncrement();
+				break;
+			} 
+			case T_AlterObjectSchemaStmt:
+			{
+				ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+					errmsg("Alter extension schema is not currently supported in Babelfish")));
```

Comment:
We need to check the object type here. There can be any object possible. Please add a test case where we execute ALTER TABLE , in this case current implementation won't throw any error.